### PR TITLE
Canonicalize stores in the CanonicalizeInstruction utility.

### PR DESCRIPTION
**Pull request by atrick**
This is the complement to load canonicalization. Although store
canonicalization is not required before diagnostics, it should be
defined in the same utility.

### CODE REVIEW

**Comment by atrick** [comment]
@swift-ci test

**Comment by atrick** [comment]
@swift-ci test

File: lib/SILOptimizer/Utils/CanonicalizeInstruction.cpp [hunk 1]
```diff
@@ -309,6 +309,84 @@ splitAggregateLoad(LoadInst *loadInst, CanonicalizeInstruction &pass) {
...
+
+  // Bail if the store's destination is not a struct_element_addr.
+  auto *sea = dyn_cast<StructElementAddrInst>(storeInst->getDest());
+  if (!sea)
+    return nextII;
+
+  auto *f = storeInst->getFunction();
+
+  // Continue up the struct_element_addr chain, as long as each struct only has
+  // a single property, creating StringInsts along the way.
```
Comment (gottesmm) [inline lib/SILOptimizer/Utils/CanonicalizeInstruction.cpp]: `StringInsts` -> `StoreInsts`

**Review by gottesmm — APPROVED** [review]
Small typo, but otherwise LGTM

**Comment by atrick** [comment]
@swift-ci smoke test and merge

**Comment by atrick** [comment]
@swift-ci smoke test and merge.

**Pull request merged**